src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,21 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books = state.books.concat(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book !== action.payload);
+    },
```

Comment:
- [ ] The `removeBook` action should filter those book that their ids are not equal to the `action.payload`. In the code above you are using filtering against the whole book object and not the ids. Kindly change it:

```JS
state.books = state.books.filter((book) => book !== action.payload);
```